rustls/src/server/tls13.rs

```diff
@@ -285,16 +282,16 @@ mod client_hello {
                 for (i, psk_id) in psk_offer.identities.iter().enumerate() {
                     let now = self.config.current_time()?;
 
-                    let resume = match self
+                    let Some(resume) = self
```

Comment:
Pre-existing nit: IMO this wasn't very readable before and hasn't gotten better (maybe even slightly worse) with let-else. Maybe tweak that?

(I usually try to avoid multi-line scrutinees because they make it harder for me to make sense of the control flow. There are a few more examples in this commit...)

Reply:
added a refactoring commit for this

Reply:
Nice!